src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -209,6 +213,10 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             if (Href != null)
             {
                 output.CopyHtmlAttribute(HrefAttributeName, context);
+
+                Href = UrlHelper.Content(Href.Trim());
```

Comment:
Add a comment about later `Href` comparison to avoid someone "cleaning this up" later.